macOS/MoltenVK issue on SDL 2.0.12

My application runs with sdl2-sys 0.34.2 but not 0.34.3, when using bundled/static-link. The error message:
```
thread 'main' panicked at 'Failed to create window: SdlError("Vulkan support is either not configured in SDL or not available in current SDL video driver (cocoa) or platform")', engine/src/init.rs:51:18
```

It may be related to this:
https://github.com/KhronosGroup/MoltenVK/issues/846

It sounds like VK_MVK_macos_surface was replaced with VK_EXT_metal_surface. I'll investigate this more later, but wanted to mention it.

I was able to build SDL2 manually and static link it. It was not an easy process so a few things I had to do in build.rs for rust-sdl2:
 * Add the Metal framework: `println!("cargo:rustc-link-lib=framework=Metal");`
 * I did not have SDL2main and didn't figure out how to generate it so I REMOVED `println!("cargo:rustc-link-lib=static=SDL2main");`
 * Add clang_rt.osx
    * LIB: `println!("cargo:rustc-link-lib=clang_rt.osx");`
    * Search path: `println!("cargo:rustc-link-search={}", "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.3/lib/darwin");` to solve this error:

```
error: linking with `cc` failed: exit code: 1
...
  = note: Undefined symbols for architecture x86_64:
            "___isPlatformVersionAtLeast", referenced from:
                _METAL_CreateRenderer in libsdl2_sys-4e45691694f9156d.rlib(SDL_render_metal.o)
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

After static linking, my vulkan app works fine. I only reproduce the above behavior if I manually set SDL_VIDEO_VULKAN to 0. I think the next step is figuring out why bundled 2.0.12 SDL2 is producing different results than static linking my own SDL2 I built manually.

--- a/changelog.md
+++ b/changelog.md
@@ -1,6 +1,10 @@
 In this file will be listed the changes, especially the breaking ones that one should be careful of
 when upgrading from a version of rust-sdl2 to another.
 
+### Unreleased
+
+ * Add patch to fix metal detection (https://bugzilla.libsdl.org/show_bug.cgi?id=4988)
+
 ### v0.34.3
 
 [PR #1027](https://github.com/Rust-SDL2/rust-sdl2/pull/1027): upgrade "bundled" version of SDL2 to 2.0.12

--- a/sdl2-sys/build.rs
+++ b/sdl2-sys/build.rs
@@ -154,6 +154,10 @@ fn patch_sdl2(sdl2_source_path: &Path) {
         // https://bugzilla.libsdl.org/show_bug.cgi?id=5105
         // Expected to be fixed in 2.0.14
         ("SDL2-2.0.12-sndio-shared-linux.patch", include_str!("patches/SDL2-2.0.12-sndio-shared-linux.patch")),
+
+        // https://bugzilla.libsdl.org/show_bug.cgi?id=4988
+        // Expected to be fixed in 2.0.14
+        ("SDL2-2.0.12-metal-detection-macos-ios.patch", include_str!("patches/SDL2-2.0.12-metal-detection-macos-ios.patch"))
     ];
     let sdl_version = format!("SDL2-{}", LASTEST_SDL2_VERSION);
 

--- a/sdl2-sys/patches/SDL2-2.0.12-metal-detection-macos-ios.patch
+++ b/sdl2-sys/patches/SDL2-2.0.12-metal-detection-macos-ios.patch
@@ -1,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 1f086fb70..0f2e74efb 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1667,7 +1667,7 @@ elseif(APPLE)
+ 
+     if(VIDEO_VULKAN OR VIDEO_METAL OR RENDER_METAL)
+       set(ORIG_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+-      set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -x objective-c")
++      set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -ObjC")
+       check_c_source_compiles("
+         #include <AvailabilityMacros.h>
+         #import <Metal/Metal.h>

